### PR TITLE
Check internal on http-cache

### PR DESCRIPTION
**Pull request by BoD**
- Make `DiskLruCache` internal
- move FileSystem out of internal package (as it is actually part of the public API of `CachingHttpEngine`)
- fix `ClassNames`

### CODE REVIEW

**Comment by martinbonnin** [comment]
> move FileSystem out of internal package (as it is actually part of the public API of CachingHttpEngine)

Can we use okio3 `Filesystem` instead of having our own version? That'd allow to leverage all the okio `FileSystems` if needed and also that'd potentially remove some bytecode

File: apollo-http-cache/api/apollo-http-cache.api [hunk 1]
```diff
@@ -62,107 +78,3 @@ public final class com/apollographql/apollo3/cache/http/HttpFetchPolicy : java/l
...
-	public abstract fun deleteContents (Ljava/io/File;)V
-	public abstract fun exists (Ljava/io/File;)Z
-	public abstract fun rename (Ljava/io/File;Ljava/io/File;)V
-	public abstract fun sink (Ljava/io/File;)Lokio/Sink;
-	public abstract fun size (Ljava/io/File;)J
-	public abstract fun source (Ljava/io/File;)Lokio/Source;
-}
-
-public final class com/apollographql/apollo3/cache/http/internal/FileSystem$Companion {
-}
```
Comment (martinbonnin) [inline apollo-http-cache/api/apollo-http-cache.api:167]: 👍

File: apollo-http-cache/src/main/kotlin/com/apollographql/apollo3/cache/http/internal/DiskLruCache.kt [hunk 2]
```diff
@@ -127,8 +126,10 @@ class DiskLruCache internal constructor(/*
      */
     val fileSystem: FileSystem,
     /** Returns the directory where this cache stores its data.  */
-    val directory: File, private val appVersion: Int, valueCount: Int, maxSize: Long,
-    executor: Executor) : Closeable, Flushable {
+    val directory: File,
+    private val appVersion: Int, valueCount: Int, maxSize: Long,
+    executor: Executor,
+) : Closeable, Flushable {
```
Comment (martinbonnin) [inline apollo-http-cache/src/main/kotlin/com/apollographql/apollo3/cache/http/internal/DiskLruCache.kt:132]: Can `directory`, `fileSystem` be private?

File: apollo-http-cache/build.gradle.kts [hunk 3]
```diff
@@ -4,6 +4,7 @@ plugins {
 
 dependencies {
   api(groovy.util.Eval.x(project, "x.dep.okHttp.okHttp"))
+  api(groovy.util.Eval.x(project, "x.dep.okio"))
```
Comment (martinbonnin) [inline apollo-http-cache/build.gradle.kts:7]: Jus curious: doesn't okhttp already has okio as an `api` dependency? Doesn't harm to add it again but that feels superfluous

File: apollo-http-cache/api/apollo-http-cache.api [hunk 4]
```diff
@@ -35,22 +35,6 @@ public final class com/apollographql/apollo3/cache/http/DiskLruHttpCache {
...
-	public static final field SYSTEM Lcom/apollographql/apollo3/cache/http/FileSystem;
-	public abstract fun appendingSink (Ljava/io/File;)Lokio/Sink;
-	public abstract fun delete (Ljava/io/File;)V
-	public abstract fun deleteContents (Ljava/io/File;)V
-	public abstract fun exists (Ljava/io/File;)Z
-	public abstract fun rename (Ljava/io/File;Ljava/io/File;)V
-	public abstract fun sink (Ljava/io/File;)Lokio/Sink;
-	public abstract fun size (Ljava/io/File;)J
-	public abstract fun source (Ljava/io/File;)Lokio/Source;
-}
```
Comment (martinbonnin) [inline apollo-http-cache/api/apollo-http-cache.api]: Nice 👍

File: apollo-http-cache/src/test/kotlin/com/apollographql/apollo3/cache/http/internal/FaultyFileSystem.kt [hunk 5]
```diff
@@ -15,94 +15,98 @@
...
+    file.parentFile.mkdirs()
+    file.createNewFile()
+  }
+  return sink(file.toOkioPath())
+}
+
+fun FaultyFileSystem.rename(from: File, to: File) = atomicMove(from.toOkioPath(), to.toOkioPath())
+fun FaultyFileSystem.delete(file: File) = delete(file.toOkioPath())
+fun FaultyFileSystem.deleteRecursively(file: File) = deleteRecursively(file.toOkioPath())
+fun FaultyFileSystem.source(file: File) = source(file.toOkioPath())
```
Comment (martinbonnin) [inline apollo-http-cache/src/test/kotlin/com/apollographql/apollo3/cache/http/internal/FaultyFileSystem.kt:111]: internal everywhere? Maybe even private?

File: apollo-http-cache/build.gradle.kts (hunk 3 above)
Comment (BoD) [reply apollo-http-cache/build.gradle.kts:7]: I think IJ couldn't resolve the imports - but I will check again.

Comment (BoD) [reply apollo-http-cache/build.gradle.kts]: No you're right it works, I guess it was a fluke :)

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**